packages/@react-spectrum/tag/test/TagGroup.test.js

```diff
@@ -58,19 +82,339 @@ describe('TagGroup', function () {
+    expect(tagDisabled).toHaveAttribute('tabIndex', '-1');
+    let tagRowNotDisabled = tagGroup.children[0];
+    let tagNotDisabled = tagRowNotDisabled.children[0];
+    act(() => {tagNotDisabled.focus();});
+    expect(tagNotDisabled).toHaveAttribute('role', 'gridcell');
+    expect(tagNotDisabled).toHaveAttribute('tabIndex', '0');
+  });
+
+  it.each`
+    Name                                                   | props                                         | orders
+    ${'(left/right arrows, ltr + horizontal) TagGroup'} | ${{locale: 'de-DE'}}                          | ${[{action: () => {userEvent.tab();}, index: 0}, {action: pressArrowRight, index: 1}, {action: pressArrowLeft, index: 0}, {action: pressArrowLeft, index: 0}]}
+    ${'(left/right arrows, rtl + horizontal) TagGroup'} | ${{locale: 'ar-AE'}}                          | ${[{action: () => {userEvent.tab();}, index: 0}, {action: pressArrowRight, index: 0}, {action: pressArrowLeft, index: 1}, {action: pressArrowLeft, index: 2}]}
+    ${'(up/down arrows, ltr + horizontal) TagGroup'}    | ${{locale: 'de-DE'}}                          | ${[{action: () => {userEvent.tab();}, index: 0}, {action: pressArrowDown, index: 1}, {action: pressArrowUp, index: 0}, {action: pressArrowUp, index: 0}]}
+    ${'(up/down arrows, rtl + horizontal) TagGroup'}    | ${{locale: 'ar-AE'}}                          | ${[{action: () => {userEvent.tab();}, index: 0}, {action: pressArrowDown, index: 1}, {action: pressArrowUp, index: 0}, {action: pressArrowUp, index: 0}]}
+  `('$Name shifts button focus in the correct direction on key press', function ({Name, props, orders}) {
+    let tree = render(
+      <Provider theme={theme} locale={props.locale}>
+        <TagGroup aria-label="tag group">
+          <Item key="1" aria-label="Tag 1">Tag 1</Item>
+          <Item key="2" aria-label="Tag 2">Tag 2</Item>
+          <Item key="3" aria-label="Tag 3">Tag 3</Item>
+        </TagGroup>
+      </Provider>
+    );
+
+    let tags = tree.getAllByRole('gridcell');
+    orders.forEach(({action, index}, i) => {
+      action(document.activeElement);
+      expect(document.activeElement).toBe(tags[index]);
+    });
+  });
+
+  it('TagGroup allows aria-label', function () {
+    let {getByRole} = render(
+      <TagGroup aria-label="tag group" disabledKeys={['3']}>
+        <Item key="1" aria-label="Tag 1">Tag 1</Item>
+      </TagGroup>
+    );
+
+    let tagGroup = getByRole('grid');
+    expect(tagGroup).toHaveAttribute('aria-label', 'tag group');
+  });
+
+  it('TagGroup allows aria-labelledby', function () {
+    let {getByRole} = render(
+      <TagGroup aria-labelledby="tag group" disabledKeys={['3']}>
+        <Item key="1" aria-label="Tag 1">Tag 1</Item>
+      </TagGroup>
+    );
+
+    let tagGroup = getByRole('grid');
+    expect(tagGroup).toHaveAttribute('aria-labelledby', 'tag group');
+  });
+
+  it('TagGroup allows aria-label on Item', function () {
```

Comment:
I tried to emulate actiongroup (since they have similar structures). one of the action stories has aria-label on the `Item` so i added it as a test for taggroup. although i don't have any stories with aria-label in `Item`...so i am happy to remove this if it's not suppose to be supported

Reply:
Will address/continue conversation in followup PR